staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2502,6 +2502,16 @@ type KubeVirtConfiguration struct {
 	// +nullable
 	// +kubebuilder:validation:Enum=Stage;LiveUpdate
 	VMRolloutStrategy *VMRolloutStrategy `json:"vmRolloutStrategy,omitempty"`
+
+	// CommonInstancetypesDeployment controls the deployment of common-instancetypes resources
+	// +nullable
+	CommonInstancetypesDeployment *CommonInstancetypesDeployment `json:"commonInstancetypesDeployment,omitempty"`
+}
+
+type CommonInstancetypesDeployment struct {
+	// Enabled controls the deployment of common-instancetypes resources, defaults to True.
+	// +nullable
+	Enabled *bool `json:"enabled,omitempty"`
 }
```

Comment:
Same as before. We reached an agreement on which if the spec is present like: `spec.CommonInstancetypesDeployment: {}` that would mean that the feature is enabled and if not present at all, the feature is disabled. See: https://github.com/kubevirt/community/pull/316#discussion_r1732862882

Reply:
That makes sense for features that default to disabled, however what about cases such as this where we want the feature to default to enabled but still allow it to be explicitly disabled by an admin?

Reply:
In such cases, IMHO, the spec should be present by default and if the admin wants to disable it, he would need to delete the spec.

Reply:
I'm not sure that behaviour is technically possible without mutating the spec and thus breaking any declarative management of the KubeVirt object?

Reply:
For anyone maintaining the KV object, not just operators.

Lets talk an example spec without `commonInstancetypesDeployment` provided:

```
spec:
 [..]
```

What I think you're suggesting is to then have that mutated with a default if it isn't set right?

```
spec:
  commonInstancetypesDeployment: {}
  [..]
```

My issue with this is that it breaks declarative management of the KV object *and* doesn't allow a user to express that they want `commonInstancetypesDeployment` disabled.

Reply:
I think I got the point. IMHO, if they maintain their own KV configuration and does not express `commonInstancetypesDeployment`, the feature shouldn't be enabled. What I mean with "the spec should be present by default...", is that in a fresh installation the feature should be enabled. However, if the user/operator decides to create/maintain their own configuration, it is up to the user to add this specific configuration to the KV object. In any case, I didn't mean to mutate KV object to add this configuration. Hope it makes sense.

Reply:
That just isn't the target behaviour here, we want to enable `commonInstancetypesDeployment` by default without a user having to request it. We should also still allow users to explicitly enable or disable and that's just not possible using only an empty struct.

Reply:
Could you please step in the design document and reflect this scenario? If this is the case, maybe we should move towards to this approach instead of empty structs.

Reply:
Empty structs are fine for disabled by default features, this is just another use case that we could also document best practice for. Is the empty structs discussion not being broken out into a standalone design btw?

Reply:
No, it is not. It is a good idea to break it down, for sure.